lib/public/AppFramework/Http/Attribute/RequestHeader.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCP\AppFramework\Http\Attribute;
+
+use Attribute;
+
+/**
+ * This attribute allows documenting request headers and is primarily intended for OpenAPI documentation.
```

Comment:
```suggestion
 * This attribute allows documenting request headers and is primarily intended for OpenAPI documentation.
 * It should be added to your controller routes when those request a header being set on the incoming request.
 * You should still check yourself that the header is set, but this attribute will make the requirement clear in th OpenAPI documentation.
```
Something like that then?